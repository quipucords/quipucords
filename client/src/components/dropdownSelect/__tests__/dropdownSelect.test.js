import React from 'react';
import { mount } from 'enzyme';
import { MenuItem } from 'patternfly-react';
import DropdownSelect from '../dropdownSelect';

describe('DropdownSelect Component', () => {
  it('should render a basic component', () => {
    const props = {
      id: 'test',
      options: [{ title: 'lorem', value: 'ipsum' }, { title: 'hello', value: 'world' }]
    };

    let component = mount(
      <DropdownSelect {...props}>
        <MenuItem eventKey="ipsum">Lorem ipsum</MenuItem>
      </DropdownSelect>
    );

    expect(component.render()).toMatchSnapshot('basic dropdown');

    props.selectValue = ['world', 'ipsum'];
    props.multiselect = true;

    component = mount(
      <DropdownSelect {...props}>
        <MenuItem eventKey="ipsum">Lorem ipsum</MenuItem>
      </DropdownSelect>
    );

    expect(component.render()).toMatchSnapshot('multiselect dropdown');
  });

  it('should allow a alternate array and object options', () => {
    const props = {
      id: 'test',
      options: ['lorem', 'ipsum', 'hello', 'world'],
      selectValue: ['ipsum']
    };

    let component = mount(
      <DropdownSelect {...props}>
        <MenuItem eventKey="ipsum">Lorem ipsum</MenuItem>
      </DropdownSelect>
    );

    expect(component.render()).toMatchSnapshot('string array');

    props.options = { lorem: 'ipsum', hello: 'world' };

    component = mount(
      <DropdownSelect {...props}>
        <MenuItem eventKey="ipsum">Lorem ipsum</MenuItem>
      </DropdownSelect>
    );

    expect(component.render()).toMatchSnapshot('key value object');
  });
});
