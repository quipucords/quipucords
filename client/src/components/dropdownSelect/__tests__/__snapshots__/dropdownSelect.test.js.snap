// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic dropdown 1`] = `
<div
  class="quipucords-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      Select option
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        lorem
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        hello
      </a>
    </li>
  </ul>
</div>
`;

exports[`key value object 1`] = `
<div
  class="quipucords-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      Select option
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        ipsum
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        world
      </a>
    </li>
  </ul>
</div>
`;

exports[`multiselect dropdown 1`] = `
<div
  class="quipucords-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      lorem,hello
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="quipucords-dropdownselect-menuitem row"
        >
          <div
            class="quipucords-dropdownselect-menuitemname col-xs-10"
          >
            lorem
          </div>
          <div
            class="quipucords-dropdownselect-menuitemcheck col-xs-2"
          >
            <span
              aria-hidden="true"
              class="fa fa-check"
            />
          </div>
        </div>
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        <div
          class="quipucords-dropdownselect-menuitem row"
        >
          <div
            class="quipucords-dropdownselect-menuitemname col-xs-10"
          >
            hello
          </div>
          <div
            class="quipucords-dropdownselect-menuitemcheck col-xs-2"
          >
            <span
              aria-hidden="true"
              class="fa fa-check"
            />
          </div>
        </div>
      </a>
    </li>
  </ul>
</div>
`;

exports[`string array 1`] = `
<div
  class="quipucords-dropdownselect  dropdown btn-group"
>
  <button
    aria-expanded="false"
    aria-haspopup="true"
    class="dropdown-toggle btn btn-default"
    id="test"
    role="button"
    type="button"
  >
    <span>
      ipsum
    </span>
     
    <span
      class="caret"
    />
  </button>
  <ul
    aria-labelledby="test"
    class="dropdown-menu"
    role="menu"
  >
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        lorem
      </a>
    </li>
    <li
      class="active"
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        ipsum
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        hello
      </a>
    </li>
    <li
      class=""
      role="presentation"
    >
      <a
        href="#"
        role="menuitem"
        tabindex="-1"
      >
        world
      </a>
    </li>
  </ul>
</div>
`;
